src/rest-client.ts

```diff
@@ -1080,76 +1476,37 @@ export class RestClient extends BaseRestClient {
     return this.get('/api/v5/market/books', { instId, sz });
   }
 
-  getCandles(
-    instId: string,
-    bar: string = '1m',
-    pagination?: Pagination
-  ): Promise<Candle[]> {
-    return this.get('/api/v5/market/candles', {
-      instId,
-      bar,
-      ...pagination,
-    });
+  getFullOrderBook(params: {
+    instId: string;
+    sz?: string;
+  }): Promise<OrderBook[]> {
+    return this.get('/api/v5/market/books-full', params);
   }
 
-  getHistoricCandles(
-    instId: string,
-    bar: string = '1m',
-    pagination?: Pagination
-  ): Promise<Candle[]> {
-    return this.get('/api/v5/market/history-candles', {
-      instId,
-      bar,
-      ...pagination,
-    });
+  getCandles(params: CandleRequest): Promise<Candle[]> {
```

Comment:
While I agree the object notation is better, this is unfortunately a breaking change to an existing function. Careful with these, there's quite a few here and I don't think it's the time to do them while also adding so many new endpoints...

Reply:
I'm conflicted because I do prefer the object notation. It's safer in the long run. Maybe we can encourage users to migrate to the object notation without breaking the existing implementation:
```
  /**
   * @deprecated this method's parameters will change to an object in the next release. Use getCandlesV2 instead.
   *
   * @param instId
   * @param bar
   * @param pagination
   * @returns
   */
  getCandles(
    instId: string,
    bar: string = '1m',
    pagination?: Pagination
  ): Promise<Candle[]> {
    return this.get('/api/v5/market/candles', {
      instId,
      bar,
      ...pagination,
    });
  }

  getCandlesV2(params: CandleRequest): Promise<Candle[]> {
    return this.get('/api/v5/market/candles', params);
  }
```